Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Projectiles.xml

```diff
@@ -0,0 +1,256 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+	<Operation Class="PatchOperationFindMod">
+		<mods>
+		<li>Alpha Animals</li>
+		</mods>
+
+		<match Class="PatchOperationSequence">
+			<operations>
+
+			<li Class="PatchOperationReplace">
+				<xpath>/Defs/ThingDef[defName="AA_AcidicWeb" or
+				defName="AA_Bullet_Barb" or
+				defName="AA_Bullet_VenomBarb" or
+				defName="AA_FireWeb" or
+				defName="AA_FrostWeb" or
+				defName="AA_Plasma" or
+				defName="AA_PoisonBolt" or
+				defName="AA_Quill" or
+				defName="AA_RedPoisonBolt" or
+				defName="AA_Web" or
+				defName="AA_FireSpit"
+				]/thingClass</xpath>
+				<value>
+					<thingClass>CombatExtended.BulletCE</thingClass>

```

Comment:
> I'm not sure you can just do that. Most of those bullets have custom effects that won't be replicated if you change it to the generic `CombatExtended.BulletCE` class.

Isn't that behaviour attached to the DamageDef itself, so it wouldn't matter since the BulletCE class has a <damageDef> node?

Reply:
Normally yes, but there are some projectiles with two effects, for example `AA_FrostWeb` stuns, but also causes frostbite.

Reply:
Would a viable patch consist of making it have secondary damage (like HE bullets) with the corresponding hediff? I'm not sure how to go around this without basically any C# knowledge.

I used the BulletCE thingClass to make armor be relevant against these ranged attacks

Reply:
Yes, actually. Just checked. Using `secondaryDamage` with the correct damage types should do the trick perfectly.

Reply:
You can find which damage type to apply to which projectile type in lines 34-60 of the C# file I've linked to above.

Reply:
Okay, I'll get into it. Some people also told me that their ranged attacks don't have any accuracy whatsoever, so I'll append some RaceDefs' `AimingAccuracy` statBase to races that use them.

This is under the premise "animals" in general should be proficient with their ranged attacks (going by DnD's logic, unarmed attacks are done with proficiency, we could go on and forth about instinct and stuff), so I'm considering putting a circa 90% accuracy to them.

@Breadbox2k19 reported these statBases actually factor in their accuracy, since as of now it seems they can't hit the broad side of a barn consistently.

Reply:
Balance is less my forte, but the aiming accuracy base is the correct one to modify for this purpose.